laalaa/apps/advisers/models.py

```diff
@@ -156,14 +157,47 @@ def __iter__(self):
 
 
 IMPORT_STATUSES = Choices(
-    ("RUNNING", "running"), ("SUCCESS", "success"), ("FAILURE", "failure"), ("ABORTED", "aborted")
+    ("CREATED", "created"),
+    ("RUNNING", "running"),
+    ("SUCCESS", "success"),
+    ("FAILURE", "failure"),
+    ("ABORTED", "aborted"),
 )
 
 
 class Import(models.Model):
     task_id = models.CharField(max_length=50, default="")
-    started = models.DateTimeField(auto_now_add=True)
+    created = models.DateTimeField(auto_now_add=True, null=True)
+    started = models.DateTimeField(auto_now_add=False, null=True)
+    completed = models.DateTimeField(auto_now_add=False, null=True)
     status = models.CharField(max_length=7, choices=list(IMPORT_STATUSES))
     filename = models.TextField()
     user = models.ForeignKey(User, null=True, on_delete=models.CASCADE)
     failure_reason = models.TextField(null=True)
+
+    def abort(self):
+        self.status = IMPORT_STATUSES.ABORTED
+        self.save(update_fields=["status"])
+
+    def start(self):
+        self.status = IMPORT_STATUSES.RUNNING
+        self.started = timezone.now()
+        self.save(update_fields=["status", "started"])
+
+    def complete(self):
+        self.status = IMPORT_STATUSES.SUCCESS
+        self.completed = timezone.now()
+        self.save(update_fields=["status", "completed"])
+
+    def is_in_progress(self):
+        return self.status in [IMPORT_STATUSES.CREATED, IMPORT_STATUSES.RUNNING]
+
+    @staticmethod
+    def get_last():
+        return Import.objects.all().order_by("-id").first()
+
+    @staticmethod
+    def abort_last():
```

Comment:
I think I'd prefer this to be on the Manager class, but do you think that's overcomplicating it?